keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+        
+        For example: consider a pod with multiple containers:
+          1. Container 1: Moderate CPU and memory usage, but occasional spikes.
+          2. Container 2: Low and consistent resource usage.
+
+
+          ```yaml
+          apiVersion: v1
+          kind: Pod
+          ...
+          spec:
+            resources:
+              requests:
+                memory: 100Mi
+                cpu: 1
+              limits:
+                memory: 400Mi
+                cpu: 2.5
+            containers:
+            - name: container-1
+                resources:
+                  requests:
+                    memory: 60Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 300Mi
+                    cpu: 2  
+            - name: container-2
+                resources:
+                  requests:
+                    memory: 40Mi
+                    cpu: 0.5
+                  limits:
+                    memory: 200Mi
+                    cpu: 1 
+          ```
+
+          The pod-level limit will serve as an aggregate ceiling. This will cover the scenario where the sum of peak usage for individual containers in a pod is high, but these
+          peaks don’t necessarily occur simultaneously.
+
+          If we simply set the pod-level limit as the sum of the individual container limits, we might overallocate resources. The pod might never actually use all those resources
+          at once, leading to wasted capacity on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < pod.spec.containers[i].resources.limits[memory/cpu]` is invalid.
+
+        A single container’s limit should not be allowed to exceed the pod level limit.
+
+#### Scheduler Changes
+
+The scheduler logic needs to be updated to support pod-level requests and limits.
+
+* **Scenario 1: Only container-level resources are specified**
+
+  No changes, same as existing.
+
+* **Scenario 2: Only pod-level resources are specified**
+
+  Consider pod-level requests to find the suitable node for pod placement.
+
+* **Scenario 3: Both container-level and pod-level resources are specified**
+
+  Scheduler will prioritize pod-level requests as the sole indicator of the pod’s resources needs. It will ignore container level requests. This ensures that the pod 
+  is scheduled on a node that can fulfill the overall resource requirements of the pod, regardless of the individual container requests.
+
+
+#### QoS Changes
+
+The QoS definitions need to be updated to consider pod-level resources. To attach a QoS class to a pod, the  pod’s resource specifications are evaluated in order:
+
+1. The values for resources specified in the pod’s `resources` stanza.
+2. The value of resources specified in the container’s `resources` stanza.
+
+In simple words, if resources are specified at pod-level, the values in pod-level `resources stanza` are given a preference in determining the QoS class.
```

Comment:
if pod limit is not set, will the sum of container limits be used instead? Or pod requests and limits must be set so the pod can become guaranteed?

Reply:
It depends on QoS. For guaranteed, pod limit should be something like:

Pod_limit = max(init container's limit, sum of app containers' limit) + sidecar container limit + pod overhead

Reply:
If pod level `resources` is set, the container-level `resources` won't be considered. I reframed this section to add more clarity. PTAL